client_generic/ContentDownloader/SheepGenerator.cpp

```diff
@@ -302,6 +302,13 @@ bool	SheepGenerator::getControlPoints( SheepUploader *uploader )
 																					1.0,
 																					1.0 );
 
+	// Convert "r3d0.sheepserver.net" to "v3d0.sheepserver.net" 
```

Comment:
We can do it in this way:

#define URL_HTTP "http://"
#define URL_HTTPS "https://"

#define CHAR_REGISTERED 'r'
#define CHAR_UNREGISTERED 'v'

const char strHttp[] = URL_HTTP;
 const char *strTarget = strHttp;
 char *strFound = std::strstr(url, strTarget);
 
 if ((NULL == strFound) || (strFound != url))
 {
  const char strHttps[] = URL_HTTPS;
  strTarget = strHttps;
  strFound = std::strstr(url, strTarget);
 }

 if ((NULL != strFound) && (strFound == url) && (CHAR_REGISTERED == strFound[strlen(strTarget)]) )
 {
  strFound[strlen(strTarget)] = CHAR_UNREGISTERED;
 }